src/reducetensor.cpp

```diff
@@ -568,10 +563,13 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
 
     if(need_indices && workspace != nullptr)
     {
-        std::size_t aTypeSize = detail::GetDataTypeSize(aDesc.GetType());
+        std::size_t aTypeSize     = detail::GetDataTypeSize(aDesc.GetType());
+        auto ws_buf12_sizeInBytes = detail::IsDynamicReductionEnabled()
+                                        ? workspaceSizeInBytes - 4096
+                                        : workspaceSizeInBytes;
 
         long byteOffset =
-            static_cast<long>((workspaceSizeInBytes / (aTypeSize + sizeof(int))) * aTypeSize);
+            static_cast<long>((ws_buf12_sizeInBytes / (aTypeSize + sizeof(int))) * aTypeSize);
```

Comment:
If I read your kernel correctly, the kernel is not packing a tensor value with its index in memory. So this way of calculating offset for index buffer is dangerous because you are assuming the data will be packed in memory byte-by-byte.


Please calculate ```byteOffset ``` for index buffer by properly calculating the buffer size for partial reduction output

Reply:
workspaceSizeInBytes is the the user allocated bytes size, it should be no less than the actual required size,  and it could be obviously bigger than the required size.   So my calculation is based on this this allocated size.  No need to worry,  I added some gap when calculating the required size, and the index buffer is aligned in 64-byte boundary.  This logic is shared by both static and dynamic reduction,  had better not be changed

Reply:
Change!